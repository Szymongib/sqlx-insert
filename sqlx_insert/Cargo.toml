[package]
name = "sqlx_insert"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
sqlx_insert_derive = { path = "../sqlx_insert_derive" }
async-trait = "0.1.52"
sqlx = { version = "0.5.10" }

[dev-dependencies]
tokio = { version = "1.16.1", features = ["full"]}
sqlx = { version = "0.5.10", features = [ "runtime-tokio-rustls", "postgres" ] }
trybuild = "1.0.54"

