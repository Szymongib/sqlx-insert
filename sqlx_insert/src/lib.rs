use async_trait::async_trait;
use sqlx;
pub use sqlx_insert_derive::SQLInsert;

#[async_trait]
pub trait SQLInsert<DB: sqlx::Database> {
    async fn sql_insert<'c, E: sqlx::Executor<'c, Database = DB>>(
        &self,
        connection: E,
    ) -> ::sqlx::Result<()>;
}
