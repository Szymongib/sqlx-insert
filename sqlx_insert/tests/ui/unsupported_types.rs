use sqlx_insert::SQLInsert;
use sqlx::Postgres;

#[derive(SQLInsert, Clone, Debug)]
#[sqlx_insert(database(Postgres))]
pub enum Thing{
    A, B, C
}

fn main() {}
