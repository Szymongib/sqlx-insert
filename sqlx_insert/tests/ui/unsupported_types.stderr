error: proc-macro derive panicked
 --> tests/ui/unsupported_types.rs:5:10
  |
5 | #[derive(SQLInsert, Clone, Debug)]
  |          ^^^^^^^^^
  |
  = help: message: failed to expand SQLInsert macro: Error("enums are not supported")

warning: unused import: `async_trait::async_trait`
 --> tests/ui/unsupported_types.rs:2:5
  |
2 | use async_trait::async_trait;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `sqlx::Postgres`
 --> tests/ui/unsupported_types.rs:3:5
  |
3 | use sqlx::Postgres;
  |     ^^^^^^^^^^^^^^
